blob/driver/driver.go

```diff
@@ -42,10 +42,10 @@ type Error interface {
 type Reader interface {
 	io.ReadCloser
 
-	// Attrs returns the object's metadata. It may return a different pointer each
-	// time, but it must return the exact same values each time it is called. The
-	// caller must not modify any fields in the returned ObjectAttrs.
-	Attrs() *ObjectAttrs
+	// ContentType is the MIME type of the blob object. It must not be empty.
+	ContentType() string
```

Comment:
Making these be methods makes introducing a field be more complicated to roll out. I suggest having two different structs: one that is the "reader attributes" and one that is "all attributes". The latter may embed the former.

Reply:
Great point, thanks. Done.